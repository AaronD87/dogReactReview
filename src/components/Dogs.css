.li {
  list-style: none;
}

.ul {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  padding-left: 6em;
}