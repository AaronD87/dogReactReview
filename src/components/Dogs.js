import React from 'react';
import PropTypes from 'prop-types';
import Dog from './Dog';
import styles from './Dogs.css';

export default function Dogs({ dogs }) {
  const listOfDogs = dogs.map((dog, idx) => {
    return <li className={styles.li} key={idx}><Dog name={dog.name} age={dog.age} weight={dog.weight}/></li>;
  });
  return (
    <ul className={styles.ul}>
      {listOfDogs}
    </ul>
  );
}

Dogs.propTypes = {
  dogs: PropTypes.array.isRequired
};
