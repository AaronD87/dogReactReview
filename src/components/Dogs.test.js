import React from 'react';
import { shallow } from 'enzyme';
import Dogs from './Dogs';

describe('Dogs', () => {
  it('renders the Dogs component', () => {
    const dogs = [{ name: 'Billy', age: 12, weight: '70 lbs' }];
    const wrapper = shallow(<Dogs dogs={ [...dogs] }/>);

    expect(wrapper).toMatchSnapshot();
  });
});
