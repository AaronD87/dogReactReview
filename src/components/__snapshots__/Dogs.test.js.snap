// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dogs renders the Dogs component 1`] = `
<ul>
  <li
    key="0"
  >
    <Dog
      age={12}
      name="Billy"
      weight="70 lbs"
    />
  </li>
</ul>
`;
